### PR TITLE
Added exception handling for entities with errors. Added endpoint to …

### PR DESCRIPTION
**Pull request by wanjohikibui**
Added exception handling for entities with errors.
#21 and 
 Added endpoint to to load geojson for one record
https://github.com/gltn/stdm-web/issues/20

### CODE REVIEW

File: stdm_config/views.py [hunk 1]
```diff
@@ -189,25 +190,25 @@ def EntityLookupSummaries(profile, entity):
 
 def LooukupSummary(child_entity, ers):
 	parent_entity = ers[0].parent
-	#select count(*), cg.value from ko_farmer f left join ko_check_gender cg on cg.id = f.gender group by cg.value;
 	query = 'select count(*), ' + parent_entity.name + '.value from '+ child_entity.name + ' join ' + parent_entity.name + ' on ' + parent_entity.name+'.'+ers[0].parent_column + '='+ child_entity.name+'.'+ers[0].child_column+ ' group by '+ parent_entity.name + '.value;'
```
Comment (gkahiu) [inline stdm_config/views.py:193]: 1. Would suggest you use Python's f-string literal for string concatenation, using the '+' sign is not recommended. Will also make the code easier to read.
2. Check line length based on PEP8 guidelines, should be a max of 79 characters per line. You can also check for other parts of the code.

File: stdm_config/views.py [hunk 2]
```diff
@@ -189,25 +190,25 @@ def EntityLookupSummaries(profile, entity):
 
 def LooukupSummary(child_entity, ers):
```
Comment (gkahiu) [inline stdm_config/views.py:191]: Function names should be in lowercase and separated by underscore if required. See PEP8 guidelines [here](https://peps.python.org/pep-0008/#descriptive-naming-styles). Kindly also review other parts of the code to ensure conformance.

File: stdm_config/views.py [hunk 3]
```diff
@@ -217,68 +218,59 @@ def EntityDetailView(request, profile_name, short_name):
...
+			
+			for key in cursor.description:
+				columns.append(toHeader(key[0]))
+		
+		lookup_summaries = EntityLookupSummaries(prof, entity)
+		spatial_results = None
+		if has_spatial_column:
+			spatial_results = entity_geojson(entity)
+	except Exception as e:
+		print(e.args)
```
Comment (gkahiu) [inline stdm_config/views.py:258]: Should this print be there or it was for debugging?

**Comment by gkahiu** [comment]
@wanjohikibui thanks for the PR. Kindly see the comments I have included for some code sections. Kindly have a look at them. My general observations are:

1. Check and adhere to PEP8 guidelines;
2. Comments are missing especially for key classes and functions, it would be good to include these, even if just one-liners;
3. Tests are missing, it would be good to include even if just unit tests for key components. You could also integrate them with GitHub workflows for fully automated CI/CD.

Am happy to discuss further.

File: stdm_config/views.py (hunk 3 above)
Comment (wanjohikibui) [reply stdm_config/views.py:283]: To remove, should be a LOGGER instead

File: stdm_config/views.py (hunk 2 above)
Comment (wanjohikibui) [reply stdm_config/views.py:198]: Yes I'm addressing the code style in a separate branch which I'll open a  PR for

**Comment by wanjohikibui** [comment]
> @wanjohikibui thanks for the PR. Kindly see the comments I have included for some code sections. Kindly have a look at them. My general observations are:
> 
> 1. Check and adhere to PEP8 guidelines;
> 2. Comments are missing especially for key classes and functions, it would be good to include these, even if just one-liners;
> 3. Tests are missing, it would be good to include even if just unit tests for key components. You could also integrate them with GitHub workflows for fully automated CI/CD.
> 
> Am happy to discuss further.

Thanks for the feedback. I'm actively working on the code style and I'll share a PR for the same

File: stdm_config/views.py (hunk 1 above)
Comment (wanjohikibui) [reply stdm_config/views.py]: This is noted. Thanks for the tip

**Pull request merged**